exporting/pubsub/pubsub_publish.cc

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include <google/pubsub/v1/pubsub.grpc.pb.h>
+#include <grpcpp/grpcpp.h>
+#include <stdexcept>
+#include "pubsub_publish.h"
+
+#define EVENT_CHECK_TIMEOUT 50
+
+struct response {
+    grpc::ClientContext *context;
+    google::pubsub::v1::PublishResponse *publish_response;
+    size_t tag;
+    grpc::Status *status;
+
+    size_t published_metrics;
+    size_t published_bytes;
+};
+
+/**
+ * Initialize a Pub/Sub client and a data structure for responses.
+ *
+ * @param pubsub_specific_data_p a pointer to a structure with instance-wide data.
+ * @param error_message report error message to a caller.
+ * @param destination a Pub/Sub service endpoint.
+ * @param credentials_file a full path for a file with google application credentials.
+ * @param project_id a project ID.
+ * @param topic_id a topic ID.
+ * @return Returns 0 on success, 1 on failure.
+ */
+int pubsub_init(
+    void *pubsub_specific_data_p, char *error_message, const char *destination, const char *credentials_file,
+    const char *project_id, const char *topic_id)
+{
+    struct pubsub_specific_data *connector_specific_data = (struct pubsub_specific_data *)pubsub_specific_data_p;
+
+    try {
+        setenv("GOOGLE_APPLICATION_CREDENTIALS", credentials_file, 0);
+
+        std::shared_ptr<grpc::ChannelCredentials> credentials = grpc::GoogleDefaultCredentials();
+        if (credentials == nullptr) {
+            std::strncpy(error_message, "Can't load credentials", ERROR_LINE_MAX);
```

Comment:
I will give a suggestion here for we avoid future errors from other developers. I verified that until now every time you called `pubsub_init`, you took care and you reset the array, but in the future case other developers call this function without to reset, we can have a problem. Considering that all the texts stored inside `error_message` are texts with well known size, please, insider a `NULL` character after each strncpy. Thank you!

Reply:
I don't see the problem with the explicit array initialization approach as far as `pubsub_init` and other functions from `pubsub_publish.cc` aren't supposed to be used anywhere else.
But, if we really want to be on the safe side, I suggest nullifying only the last byte of the array and don't bother with the string length.

Reply:
To add NULL at the end is enough. I agree your code is not creating possible bugs, my concern is with future codes calling this function.

Reply:
Done.